bin/start

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+
+declare minikube=$(which minikube);
+declare kubectl=$(which kubectl);
+declare workdir=$(cd $(dirname $0)/.. && pwd);
+
+
+start_minikube() {
+	local sudo_cmd='sudo -E ';
+	local vmdriver='none';
+	if [[ "${OSTYPE}" == "darwin"* ]]; then
+		sudo_cmd='';
+		vmdriver='docker-machine-driver-xhyve';
+	fi
+	# set the env
+	export MINIKUBE_WANTUPDATENOTIFICATION=false;
+	export MINIKUBE_WANTREPORTERRORPROMPT=false;
+	export MINIKUBE_HOME=$HOME;
+	export CHANGE_MINIKUBE_NONE_USER=true;
+	mkdir $HOME/.kube &> /dev/null || true;
+	touch $HOME/.kube/config;
+
+	echo "[*] Starting minikube ...";
+	export KUBECONFIG=$HOME/.kube/config;
+	${sudo_cmd}${minikube} start --vm-driver=${vmdriver};
+	if [[ $? != 0 ]]; then
+		exit 2;
+	fi
+
+	echo -n "[*] Waiting for the cluster to become available ";
+	for i in {1..150}; do
+		${kubectl} get pods &> /dev/null;
+		if [[ $? != 1 ]]; then
+			break;
+		fi
+		echo -n ".";
+		sleep 2;
+	done
+	echo;
+}
+
+
+if [[ -z "${minikube}" || -z "${kubectl}" ]]; then
+	echo "minikube and/or kubectl commands not found!" > 2;
+	exit 1;
+fi
+
+${minikube} status &> /dev/null;
```

Comment:
One last thing I didn't realize before - on OS X `minikube status` returns 0 even is qinikube is not running. `minikube ip` works on the contrary.

Reply:
Hm, that weird. On Linux, `minikube ip` resolves perfectly fine even when the cluster is stopped. I can special-case this situation as well (check `ip` for OSX, and `status` for Linux).

Reply:
@Pchelolo what is output and error code on OSX when the cluster is stopped for `minikube ip` ?

Reply:
@d00rman 
```
bash-3.2$ minikube ip
E1220 13:42:06.032539   33598 ip.go:48] Error getting IP:  Host is not running
bash-3.2$ echo $?
1
bash-3.2$ minikube status
minikube: Stopped
cluster: 
kubectl: 
bash-3.2$ echo $?
1
```

Reply:
This seems contradictory to what you posted earlier. Here `minikube status` seems to report the correct exit code (`1`) if it's stopped, or am I missing something?

Reply:
Oh @d00rman `brew upgrade minikube` failed me in a lot of ways today :) Lemme try to run the script again as is